File: 2021/01/17/jh20s/contest#222/1713.py
class Solution:
    def minOperations(self, target, arr):
        arr_dic = {a:i for i ,a in enumerate(target)}
        stack = []
        for i in arr:
            if i not in arr_dic:
                continue
            bi = bisect.bisect_left(stack,arr_dic[i])
            if bi == len(stack):
                stack.append(0)
            stack[bi] = arr_dic[i]

        return len(target) - len(stack)
        

File: 2021/01/17/jh20s/contest#222/1713.cpp
class Solution {
public:
int minOperations(vector<int>& target, vector<int>& arr) {
	vector<int> v;
	map<int, int> m;
	for (int i = 0; i < target.size(); i++) {
		m[target[i]] = i+1;
	}
	for (int i = 0; i < arr.size(); i++) {
		if (m.find(arr[i]) == m.end())
			continue;
		int val = m[arr[i]];
		auto it = lower_bound(v.begin(), v.end(), val);
		if (it != v.end()) {
			v[it - v.begin()] = val;
		}
		else {
			v.push_back(val);
		}
	}
	return target.size() - v.size();
}
};